receiver/hostmetricsreceiver/internal/scraper/cpuscraper/internal/metadata/generated_metrics.go

```diff
@@ -14,84 +14,134 @@
 
 // Code generated by mdatagen. DO NOT EDIT.
 
+// This file contains an experimental metrics builder generated by "mdatagen --experimental-gen metadata.yaml" command.
+
 package metadata
 
 import (
+	"time"
+
 	"go.opentelemetry.io/collector/config"
 	"go.opentelemetry.io/collector/model/pdata"
 )
 
 // Type is the component type name.
 const Type config.Type = "cpu"
 
-// MetricIntf is an interface to generically interact with generated metric.
-type MetricIntf interface {
-	Name() string
-	New() pdata.Metric
-	Init(metric pdata.Metric)
+// MetricConfig provides common settings for a particular metric.
+type MetricConfig struct {
```

Comment:
We use "Settings" for this when ask users to embed into their config.

Reply:
Changed to Settings